tests/test_model.py

```diff
@@ -1592,3 +1605,46 @@ def func(x, k=1, b=0):
 
     assert result.rsquared < 1.00
     assert result.rsquared > 0.95
+
+
+# based on Github #953
+class PolynomialFunction:
+    def __init__(self, degree: int = 1):
+        self.degree = degree
+
+    @property
+    def __name__(self) -> str:
+        return self.__class__.__name__
+
+    @property
+    def argnames(self):   # for Py39+:  -> list[str]:
```

Comment:
I'd suggest to completely remove; I don't think there are any plans to add type annotations anytime soon...